cli/src/validate.rs

```diff
@@ -0,0 +1,261 @@
+            Ok::<(Vec<(Report, String)>, Vec<(JunitParseError, String)>), anyhow::Error>(
+                file_sets_parse_results,
+            )
+        },
+    )
+}
+
+fn print_matched_files(file_sets: &[FileSet], file_counter: FileSetCounter) {
+    log::info!("");
+    log::info!(
+        "Validating the following {} files matching the provided globs:",
+        file_counter.get_count()
+    );
+    for file_set in file_sets {
+        log::info!(
+            "  File set ({:?}): {}",
+            file_set.file_set_type,
+            file_set.glob
+        );
+        for file in &file_set.files {
+            log::info!("\t{}", file.original_path_rel);
+        }
+    }
+}
+
+fn print_parse_errors(parse_errors: Vec<(JunitParseError, String)>) {
+    log::info!("");
+    log::warn!(
+        "Encountered the following {} non-fatal errors while parsing files:",
+        parse_errors.len().to_string().yellow()
+    );
+
+    let mut current_file_original_path = parse_errors[0].1.clone();
+    log::warn!("  File: {}", current_file_original_path);
+
+    for error in parse_errors {
+        if error.1 != current_file_original_path {
+            current_file_original_path = error.1;
+            log::warn!("  File: {}", current_file_original_path);
+        }
+
+        log::warn!("\t{}", error.0);
+    }
+}
+
+fn print_summary_failure(
+    num_reports: usize,
+    num_invalid_reports: usize,
+    num_optionally_invalid_reports: usize,
+) {
+    log::info!("");
+    let num_optional_validation_errors_str = if num_optionally_invalid_reports > 0 {
+        format!(
+            ", {} files have optional validation errors",
+            num_optionally_invalid_reports.to_string().yellow()
+        )
+    } else {
+        String::from("")
+    };
+    log::info!(
+        "{} files are valid, {} files are not valid{}",
+        (num_reports - num_invalid_reports).to_string().green(),
+        num_invalid_reports.to_string().red(),
+        num_optional_validation_errors_str,
+    );
+}
+
+fn print_summary_success(num_reports: usize, num_optionally_invalid_reports: usize) {
+    log::info!("");
+    let num_optional_validation_errors_str = if num_optionally_invalid_reports > 0 {
+        format!(
+            " ({} files with optional validation errors)",
+            num_optionally_invalid_reports.to_string().yellow()
+        )
+    } else {
+        String::from("")
+    };
+
+    log::info!(
+        "All {} files are valid!{}",
+        num_reports.to_string().green(),
+        num_optional_validation_errors_str
+    );
+    log::info!(
+        "Navigate to <URL for next onboarding step> to continue getting started with Flaky Tests"
+    );
+}
+
+fn print_validation_errors(
+    report_validations: &[(JunitReportValidation, String)],
+) -> (usize, usize) {
+    log::info!("");
+    let mut num_invalid_reports: usize = 0;
+    let mut num_optionally_invalid_reports: usize = 0;
```

Comment:
nit: rename to suboptimal for consistency with the underlying types/enums
(and rename elsewhere)